### PR TITLE
Dependency Updates

### PR DESCRIPTION
**Pull request by krystophv**
An effort to update _all_ the dependencies in the Uploader to their most recent releases. A large number of them were patch/minor releases and required no code changes. A handful of the major version revision updates also didn't require code changes. 

`chai` update changes comparisons for `Error` objects from checking that the `message` property matched to being `===` referential equivalence, forcing some strange modifications to tests. 

removed the `d3` dependency as it was _only_ being used to create a test fixture and nothing else (replaced with `lodash` version of generating the fixture)

used a number of codemods to handle the `react` updates and the `lodash` updates

`react-router` and `redux-form` both had semi-complete rewrites that severely changed the paradigm of how each of them function and came with a lot of code modification. luckily we were only using `redux-form` in one place so far and the `react-router` migration was pretty straightforward (outside the use of `react-router-redux`, which is arguable as to whether or not we _should_, but it was still easier to maintain current patterns than to remove it for now)

It's going to be easier to review this one commit at a time.

### CODE REVIEW

**Comment by krystophv** [comment]
[#492 ⁃ <L> [uploader] update lodash from 3.x to 4.x](https://trello.com/c/kTDfWrzS/125-492-l-uploader-update-lodash-from-3x-to-4x)

File: app/containers/App.js [hunk 1]
```diff
@@ -117,15 +125,23 @@ export class App extends Component {
...
     var serverinfo = serverdata[info.label];
-    this.props.route.api.setHosts(serverinfo);
+    this.props.api.setHosts(serverinfo);
     this.setState({server: info.label});
   };
 
   render() {
     return (
       <div className={styles.app} onClick={this.handleDismissDropdown}>
         <Header location={this.props.location} />
```
Comment (krystophv) [inline app/containers/App.js]: Please excuse the spacing below, I got my editor sorted out and untabified everything later.

**Comment by krystophv** [comment]
Pinging for review(s). @pazaan can probably adequately review most of the updates, but it will probably be useful to have @cbwebdevelopment to review the React/Redux specific commits like the `redux-form` and `react-router` updates along with the react codemod updates. (redux-form: https://github.com/tidepool-org/chrome-uploader/pull/528/commits/bc1eac23c654978b709756efc0d252982434a334, react-router: https://github.com/tidepool-org/chrome-uploader/pull/528/commits/89f6dffcad95ec40fc923edda9a30ad7cd9a9fcb, codemods: https://github.com/tidepool-org/chrome-uploader/pull/528/commits/77e636fab63173cddf5d4de9857d708b67b20833, https://github.com/tidepool-org/chrome-uploader/pull/528/commits/db539a747861d7ed01d0ff283ca54cffe1d8c318, https://github.com/tidepool-org/chrome-uploader/pull/528/commits/c4ab3fc8facb459c9fd549f3a283404de14e0845)

File: test/app/actions/utils.test.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ describe('utils', () => {
       expect(dispatch.callCount).to.equal(0);
       fn(err);
       expect(dispatch.callCount).to.equal(1);
-      expect(dispatch.firstCall.args[0]).to.deep.equal({
+			const expectedAction = {
```
Comment (pazaan) [inline test/app/actions/utils.test.js]: Weird spacing...

File: test/app/actions/sync.test.js [hunk 3]
```diff
@@ -439,7 +439,7 @@ describe('Synchronous Actions', () => {
           payload: new Error(errorText.E_INIT),
           meta: {source: actionSources[actionTypes.INIT_APP_FAILURE]}
         };
-				const action = syncActions.initFailure(err)
+				const action = syncActions.initFailure(err);
```
Comment (pazaan) [inline test/app/actions/sync.test.js]: Weird spacing (compared to below)

File: package.json [hunk 4]
```diff
@@ -41,9 +41,10 @@
     "decompress": "4.2.0",
     "electron-debug": "1.4.0",
     "electron-is-dev": "0.3.0",
+    "eslint-plugin-lodash": "^2.4.5",
     "immutability-helper": "2.4.0",
     "is-electron": "2.1.0",
-    "lodash": "3.10.1",
+    "lodash": "4.17.4",
```
Comment (pazaan) [inline package.json]: This makes me happy 😺

File: webpack.config.production.js [hunk 5]
```diff
@@ -41,103 +40,130 @@ export default validate(merge(baseConfig, {
...
+    rules: [
       // Extract all .global.css to style.css as is
       {
         test: /\.global\.css$/,
-        loader: ExtractTextPlugin.extract(
-          'style-loader',
-          'css-loader'
-        )
+        use: ExtractTextPlugin.extract({
+					fallback: 'style-loader',
```
Comment (pazaan) [inline webpack.config.production.js]: Odd spacing?

File: webpack.config.production.js [hunk 6]
```diff
@@ -41,103 +40,130 @@ export default validate(merge(baseConfig, {
...
 
       // Pipe other styles through css modules and append to style.css
       {
         test: /^((?!\.global).)*\.css$/,
-        loader: ExtractTextPlugin.extract(
-          'style-loader',
-          'css-loader?modules&importLoaders=1&localIdentName=[name]__[local]___[hash:base64:5]'
-        )
+        use: ExtractTextPlugin.extract({
+					fallback: 'style-loader',
```
Comment (pazaan) [inline webpack.config.production.js]: Odd spacing?

File: package.json [hunk 7]
```diff
@@ -22,43 +21,45 @@
...
+    "electron-debug": "1.4.0",
     "electron-is-dev": "0.3.0",
-    "font-awesome": "4.7.0",
-    "is-electron": "2.0.0",
-    "lodash": "3.10.1",
+    "eslint-plugin-lodash": "^2.4.5",
+    "history": "4.7.2",
+    "immutability-helper": "2.4.0",
+    "is-electron": "2.1.0",
+    "lodash": "4.17.4",
```
Comment (pazaan) [inline package.json]: This makes me happy 😺

**Pull request merged**